saleor/graphql/account/utils.py

```diff
@@ -73,46 +75,74 @@ class Meta:
 
     @classmethod
     def clean_instance(cls, info, instance):
-        super().clean_instance(info, instance)
-        if not instance.is_staff:
-            raise ValidationError(
-                {
-                    "id": ValidationError(
-                        "Cannot delete a non-staff user.",
-                        code=AccountErrorCode.DELETE_NON_STAFF_USER,
-                    )
-                }
-            )
+        errors = defaultdict(list)
+        cls.check_if_users_can_be_deleted(info, [instance], "id", errors)
+        cls.check_if_requestor_can_manage_users(info, [instance], "id", errors)
+        cls.check_if_removing_left_not_manageable_permissions(
+            info, [instance], "id", errors
+        )
+        if errors:
+            raise ValidationError(errors)
 
-        cls.check_if_requestor_can_manage_user(info, instance)
-        cls.check_if_removing_left_not_manageable_permissions(instance)
+    @classmethod
+    def check_if_users_can_be_deleted(cls, info, instances, field, errors):
```

Comment:
We should add a docstring explaining that it checks that we're not deleting non-staff users here.